src/javascript/node-project.ts

```diff
@@ -93,6 +94,12 @@ export interface NodeProjectOptions
    */
   readonly buildWorkflow?: boolean;
 
+  /**
+   * Name of PR build workflow.
+   * @default "build"
+   */
+  readonly buildWorkflowName?: string;
```

Comment:
```suggestion
  readonly buildWorkflowOptions?: BuildWorkflowOptions;
```

And then define BuildWorkflowOptions

Reply:
yeah I was following the pattern of `releaseWorkflowName`. But I agree its a better pattern to define an object with that option nested inside of it

Reply:
I'm not sure it would be desirable to have two different `BuildWorkFlowOptions` types. So I'm considering just using [the existing `BuildWorkflowOptions` interface](https://github.com/projen/projen/blob/0d9027ac26d93c30784c1e184312189bc9e31cf5/src/build/build-workflow.ts#L33) here which would support a customizing a bunch of other things

Reply:
`BuildWorkflowOptions` has a few properties that shouldn't be exposed at the `NodeProject` constructor level. So I made an interface called `NodeProjectBuildWorkflowOptions`. I wanted to use `Omit` with `BuildWorkflowOptions` to just exclude a few properties from `BuildWorkflowOptions`. But jsii doesn't support `Omit` 😕 . So I copy pasted properties from `BuildWorkflowOptions`.
Since there are several other `NodeProject` options that  are specific to `BuildWorkflow`, I marked them as deprecated and they can be replaced with properties within `NodeProjectBuildWorkflowOptions`.